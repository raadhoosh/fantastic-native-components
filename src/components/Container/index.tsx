import * as React from "react";
import View from "./style.index";
type Props = {
    theme?: any,
    children: JSX.Element | JSX.Element[] | string;
};

export default (props: Props) => {
    return <View {...props} />;
};