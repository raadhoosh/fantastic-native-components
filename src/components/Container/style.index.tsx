import styled from "styled-components";
import { View as ReView } from "react-native";
const View = styled(ReView)`
flex: 1,
padding-top: ${(props: any) => props.theme.isIphoneX ? 20 : 0};
`;

export default View;