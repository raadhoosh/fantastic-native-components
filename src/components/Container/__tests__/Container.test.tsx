import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Container from '../index';
import { View, Text } from 'react-native';
it('renders correctly with defaults', () => {
  const app = renderer
    .create(<Container >
      <View>
        <Text>Test</Text>
      </View>
    </Container>)
    .toJSON();
  expect(app).toMatchSnapshot();
});