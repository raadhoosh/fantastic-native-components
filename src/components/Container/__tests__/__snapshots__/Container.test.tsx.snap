// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  className="sc-bdVaJa gGuYrZ"
>
  <View>
    <Text>
      Test
    </Text>
  </View>
</View>
`;
